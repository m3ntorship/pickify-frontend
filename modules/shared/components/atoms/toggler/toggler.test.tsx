import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Toggler from './index';

describe('Toggler Component with snapshot', () => {
  it('should render classnames include (w-7 h-4) and (opacity-25) when passing disabled={true} and size=sm ', () => {
    const tree = renderer
      .create(
        <Toggler
          size="sm"
          disabled
          checked={false}
          id="test"
          onChange={(): boolean => {
            return true;
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render classnames include (w-7 h-4) and do NOT include (opacity-25) when passing disabled={false} and size=sm ', () => {
    const tree = renderer
      .create(
        <Toggler
          size="sm"
          checked={false}
          id="test"
          onChange={(): boolean => {
            return true;
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render include (w-sm h-6) and classnames include (opacity-25) when passing disabled={true} and size=default ', () => {
    const tree = renderer
      .create(
        <Toggler
          size="default"
          disabled
          checked={false}
          id="test"
          onChange={(): boolean => {
            return true;
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render include (w-sm h-6) and classnames do NOT include (opacity-25) when passing disabled={false} and size=default ', () => {
    const tree = render(
      <Toggler
        size="default"
        checked={false}
        id="test"
        onChange={(): boolean => {
          return true;
        }}
      />,
    );
    expect(tree).toMatchSnapshot();
  });
});
describe('Toggler Component with react testing library', () => {
  it('should be clicked once', () => {
    const handleClick = jest.fn();
    render(
      <Toggler size="sm" checked={false} id="test" onChange={handleClick} />,
    );
    const checkbox = screen.getByTestId('test') as HTMLInputElement;
    const calledonce = 1;
    userEvent.click(checkbox);
    expect(handleClick).toHaveBeenCalledTimes(calledonce);
    expect(checkbox.checked).toEqual(true);
  });
});
