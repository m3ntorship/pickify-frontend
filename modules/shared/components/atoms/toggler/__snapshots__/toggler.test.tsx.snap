// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggler Component with snapshot should render classnames include (w-7 h-4) and (opacity-25) when passing disabled={true} and size=sm  1`] = `
<label
  className="undefined"
  htmlFor="test"
>
  <input
    className="opacity-0 absolute undefined"
    data-testid="test"
    defaultChecked={false}
    disabled={true}
    id="test"
    onChange={[Function]}
    type="checkbox"
  />
  <div
    className=""
  />
</label>
`;

exports[`Toggler Component with snapshot should render classnames include (w-7 h-4) and do NOT include (opacity-25) when passing disabled={false} and size=sm  1`] = `
<label
  className=""
  htmlFor="test"
>
  <input
    className="opacity-0 absolute"
    data-testid="test"
    defaultChecked={false}
    disabled={false}
    id="test"
    onChange={[Function]}
    type="checkbox"
  />
  <div
    className=""
  />
</label>
`;

exports[`Toggler Component with snapshot should render include (w-sm h-6) and classnames do NOT include (opacity-25) when passing disabled={false} and size=default  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class=""
        for="test"
      >
        <input
          class="opacity-0 absolute"
          data-testid="test"
          id="test"
          type="checkbox"
        />
        <div
          class="undefined"
        />
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class=""
      for="test"
    >
      <input
        class="opacity-0 absolute"
        data-testid="test"
        id="test"
        type="checkbox"
      />
      <div
        class="undefined"
      />
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Toggler Component with snapshot should render include (w-sm h-6) and classnames include (opacity-25) when passing disabled={true} and size=default  1`] = `
<label
  className="undefined"
  htmlFor="test"
>
  <input
    className="opacity-0 absolute undefined"
    data-testid="test"
    defaultChecked={false}
    disabled={true}
    id="test"
    onChange={[Function]}
    type="checkbox"
  />
  <div
    className="undefined"
  />
</label>
`;
